### PR TITLE
Add option to keep index in returned data frame.

### LINKED ISSUE #27: Add option to return indices
I want to do some analysis where it's useful to compare the data in my original trajectory with the simplified trajectory.

To do this, I propose to create an argument called `keep_index`.  If this is TRUE, then return a data frame with `x`, `y` and `index`:

```R
> RamerDouglasPeucker(x, y, 5, keep_index = TRUE)
      x     y index
1   3.5 21.25     1
2  23.2  3.10     3
3  54.6 18.15     5
4  71.5  9.70     7
5 101.3 21.10     8
```

### PR DESCRIPTION
**Pull request by andrie**
This resolved #27

### CODE REVIEW

File: src/wrapper.cpp [hunk 1]
```diff
@@ -60,7 +63,14 @@ Rcpp::DataFrame RamerDouglasPeucker(Rcpp::NumericVector x, Rcpp::NumericVector y
         std::size_t index = indicesToKeep[i];
         xOut[i] = x[index];
         yOut[i] = y[index];
+        indicesToKeep[i] += 1; // index from 1 before returning to R
```
Comment (robertdj) [inline src/wrapper.cpp:66]: I prefer the comment on a separate line above

**Review by robertdj — CHANGES_REQUESTED** [review]
Generally looking good! Just a small change before I accept.

File: src/wrapper.cpp (hunk 1 above)
Comment (andrie) [reply src/wrapper.cpp]: Thanks, I've applied this change.

**Comment by andrie** [comment]
I apologise for this, but once I fixed the change you requested, I realised I didn't push my most recent commit.  This adds a test and renames the argument from `keepIndex` to `keep_index` to follow `tidyverse` style guide.

**Comment by robertdj** [comment]
Well spotted -- thanks!

**Comment by robertdj** [comment]
It turns out that the GitHub Actions are outdated, causing the CMD check on macOS to fail. Let me update the Action to make sure it all works here.

**Review by robertdj — APPROVED** [review]

**Pull request merged**